### DISCUSSION TITLE
GitHub page build fails (jekyll, theme: minima, gh-pages branch)

### DISCUSSION THREAD

**Discussion by NicoleEic** [community/community#23018] [Other Feature Feedback, Questions, & Ideas]
<p>Hello GitHub community,</p>
<p>I’m having trouble publishing a site using GitHub pages and none of the forum articles seemed to provide a solution, so far. I’m a beginner, so my apologies, if this is a duplicate.</p>
<p>The source for my page is the gh-pages branch of my repository: </p>
<pre><code>https://github.com/NicoleEic/Brain_and_Code/tree/gh-pages
</code></pre>
<p>I have changed the settings to use the gh-pages branch to build the site (-&gt; Your site is published at <a href="https://nicoleeic.github.io/Brain_and_Code/" rel="nofollow noopener">https://nicoleeic.github.io/Brain_and_Code/</a>)</p>
<p><img alt=""></p>
<p><img alt=""></p>
<p>However, when I navigate to the URL shown above, I can only access the ‘home’ site, which looks like something was wrong with the layout. Also, the link to the individual posts gives me a 404 error.</p>
<p><img alt=""></p>
<p>I suspect that the problem has something to do with the theme and layout. I’m using the ‘minima’ theme, so I have the following line in my Gemfile: </p>
<pre><code>gem "minima"
</code></pre>
<p>and in my _config.yml file:</p>
<pre><code>theme: minima
</code></pre>
<p>I don’t have a _layouts subfolder, because I understood that this is not needed, when I use the default themes.</p>
<p>The local build of the page on my computer works fine: </p>
<pre><code>Brain_and_Code(gh-pages)
==&gt;$ git pull
Already up to date.

Brain_and_Code(gh-pages)
==&gt;$ bundle exec jekyll serve
Configuration file: /xxx/Brain_and_Code/_config.yml
            Source: /xxx/Brain_and_Code
       Destination: /xxx/Brain_and_Code/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
       Jekyll Feed: Generating feed for posts
                    done in 0.791 seconds.
 Auto-regeneration: enabled for '/xxx/Brain_and_Code'
    Server address: http://xxx.x.x.x:4000/
  Server running... press ctrl-c to stop.
</code></pre>
<p><img alt="">Any comments on what I might be doing wrong would be greatly appreciated!<br>
Would any additional information from my side be useful to pin down the problem?</p>
<p>Thanks,</p>
<p>Nicole</p>

**Comment by yabberish** [comment]
<p>Seems to be working fine for me.</p>

**Comment by NicoleEic** [comment, marked as answer]
<p>Thanks for looking into this!</p>
<p>Actually, with the help of <a href="https://github.com/alvinsight" rel="nofollow noopener">alvinsight</a> I found the problem, which were relative URLs.</p>
<p>The ‘baseurl’ in the _config.yml file was not defined.</p>
<p>I have changed it to:</p>
<pre><code>baseurl: "Brain_and_Code"
</code></pre>
<p>and updated all relative paths to images, etc. and everything looks fine now.</p>
